overlord/servicestate/servicestate.go

```diff
@@ -49,9 +49,31 @@ type Instruction struct {
 
 type ServiceActionConflictError struct{ error }
 
+func computeExplicitServices(appInfos []*snap.AppInfo, names []string) map[string][]string {
+	explicitServices := make(map[string][]string, len(appInfos))
+	requested := make(map[string]bool, len(names))
+	for _, name := range names {
+		// Name might also be a snap name (or other strings the user wrote on
+		// the command line), but the loop below ensures that this function
+		// considers only application names.
+		requested[name] = true
+	}
+
+	for _, app := range appInfos {
+		snapName := app.Snap.InstanceName()
+		appName := app.String()
+		if requested[appName] {
+			explicitServices[snapName] = append(explicitServices[snapName], app.Name)
```

Comment:
so this should use app.ServiceName() instead of app.Name and it gets the same kind of content as Services